src/test/java/com/example/onlinebookstore/service/BookServiceImplTest.java

```diff
@@ -0,0 +1,315 @@
+        CreateBookRequestDto createBookRequestDto = new CreateBookRequestDto();
+        createBookRequestDto.setTitle("TestBook1");
+        createBookRequestDto.setAuthor("TestAuthor1");
+        createBookRequestDto.setIsbn("TestIsbn1");
+        createBookRequestDto.setPrice(BigDecimal.valueOf(9.99));
+        createBookRequestDto.setDescription("Test Description1");
+        createBookRequestDto.setCoverImage("TestCoverImage");
+        createBookRequestDto.setCategoryIds(Set.of(1L));
+        return createBookRequestDto;
+    }
+
+    private Book getBookFromCreateBookRequestDto(CreateBookRequestDto createBookRequestDto) {
+        Book book = new Book();
+        book.setTitle(createBookRequestDto.getTitle());
+        book.setAuthor(createBookRequestDto.getAuthor());
+        book.setIsbn(createBookRequestDto.getIsbn());
+        book.setPrice(createBookRequestDto.getPrice());
+        book.setDescription(createBookRequestDto.getDescription());
+        book.setCoverImage(createBookRequestDto.getCoverImage());
+        book.setCategories(createBookRequestDto.getCategoryIds().stream()
+                .map(Category::new)
+                .collect(Collectors.toSet()));
+        return book;
+    }
+
+    private BookDto getBookDtoFromBook(Book book) {
+        BookDto bookDto = new BookDto();
+        bookDto.setTitle(book.getTitle());
+        bookDto.setAuthor(book.getAuthor());
+        bookDto.setIsbn(book.getIsbn());
+        bookDto.setPrice(book.getPrice());
+        bookDto.setDescription(book.getDescription());
+        bookDto.setCoverImage(book.getCoverImage());
+        bookDto.setCategoryIds(book.getCategories().stream()
+                .map(c -> c.getId())
+                .collect(Collectors.toSet()));
+        return bookDto;
+    }
+
+    private Book getTestBook() {
+        Book book = new Book();
+        book.setId(1L);
+        book.setTitle("TestBook1");
+        book.setAuthor("TestAuthor1");
+        book.setIsbn("TestIsbn1");
+        book.setPrice(BigDecimal.valueOf(9.99));
+        book.setDescription("Test Description1");
+        book.setCoverImage("TestCoverImage");
+        book.setCategories(Set.of(getTestCategory()));
+        return book;
+    }
+
+    private Category getTestCategory() {
+        Category category = new Category();
+        category.setId(1L);
+        category.setName("Test Name");
+        category.setDescription("Test Description");
+        return category;
+    }
+
+    private BookDtoWithoutCategoryIds getBookDtoWithoutCategoryIdsFromBook(Book book) {
+        BookDtoWithoutCategoryIds dto = new BookDtoWithoutCategoryIds();
+        dto.setId(book.getId());
+        dto.setAuthor(book.getAuthor());
+        dto.setIsbn(book.getIsbn());
+        dto.setTitle(book.getTitle());
+        dto.setDescription(book.getDescription());
+        dto.setCoverImage(book.getCoverImage());
+        dto.setPrice(book.getPrice());
+        return dto;
+    }
```

Comment:
If you use this templates in other tests classes, you can make it static and place in some util class